kong/plugins/hmac-auth/access.lua

```diff
@@ -29,6 +31,63 @@ local SIGNATURE_NOT_SAME = "HMAC signature does not match"
 
 local _M = {}
 
+local hmac = {
+  ["hmac-sha1"] = function(secret, data)
+    return crypto.hmac.digest("sha1", data, secret, true)
+  end,
+  ["hmac-sha256"] = function(secret, data)
+    return crypto.hmac.digest("sha256", data, secret, true)
+  end,
+  ["hmac-sha384"] = function(secret, data)
+    return crypto.hmac.digest("sha384", data, secret, true)
+  end,
+  ["hmac-sha512"] = function(secret, data)
+    return crypto.hmac.digest("sha512", data, secret, true)
+  end
+}
+
+local function list_as_set(list)
+  local set = {}
+  for _, v in ipairs(list) do
+    set[v] = true;
+  end
+
+  return set
+end
+
+local function validate_params(params, conf)
+  -- check username and signature are present
+  if not (params.username and params.signature) then
+    return nil, "username or signature missing"
+  end
+
+  -- check enforced headers are present
+  local enfoprced_header_set = list_as_set(conf.enforce_headers)
```

Comment:
typo, 'enfoprced_header_set'.

also, how hard would it be to cache the `enforce_headers` table, so we dont have to constantly call list_as_set?